[OSDOCS-12515]: HCP support matrix

lahinson opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):4.17+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: https://issues.redhat.com/browse/OSDOCS-12515
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://84360--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-prepare/hcp-requirements
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,67 @@
+.Supported {mce-short} versions for {product-title} management clusters
+[cols="2",options="header"]
+|===
+|Management cluster version |Supported {mce-short} version
```
https://access.redhat.com/support/policy/updates/openshift_operators#:~:text=Multi%2DCluster%20Engine we can point here for a more sustainable tracking of the versions as well.

zanetworker replied:
Yes the MCE part which is burried under the ACM one.

lahinson replied:
Got it. I wasn't able to add a link directly to that section, but I did add a link to the OCP Operator Update Information Checker: https://access.redhat.com/labs/ocpouic/?operator=multicluster-engine&&upgrade_path=4.14%20to%204.16

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,67 @@
+|4.12 - 4.17
+|4.14 - 4.17
+
+|{ibm-power-title}
+|4.17
+|4.17
+
```
This was possible as TP before 4.17

lahinson replied:
Updated to "4.15 - 4.17", as I believe 4.15 was the first release where IBM Power was TP.

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,67 @@
+|4.17
+|4.17
+
+|{ibm-z-title}
+|4.17
+|4.17
```
This was possible as TP before 4.17

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,67 @@
+|4.17
+|4.17
+
+|{VirtProductName}
+|4.14 - 4.17
+|4.14 - 4.17
```
GA since 4.17

lahinson replied:
Was this a typo?

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,67 @@
+|4.14 - 4.17
+|4.14 - 4.17
+
+|Bare metal
+|4.14 - 4.17
+|4.14 - 4.17
```
GA since 4.14

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,67 @@
+|4.14 - 4.17
+|4.14 - 4.17
+
+|Non-bare-metal agent machines (Technology Preview)
+|4.17
+|4.17
+|===
```
TP since for 4.16 IIRC

zanetworker commented:
Can we add a table for the MCE 2.n to Hosted cluster 4.y mapping? 

Here's the updated table with **4.17** marked as a current version:

---


| **Hosted Cluster Version (4.y) that (HO of) MCE 2.n Can Create** | **MCE 2.4** | **MCE 2.5** | **MCE 2.6** | **MCE 2.7** | **MCE 2.8 (Future)** |
|------------------------------------------------------------------|-------------|-------------|-------------|----------------------|----------------------|
| **4.14**                                                         | Y           | Y           | Y           | Y                    | Y                    |
| **4.15**                                                         | N           | Y           | Y           | Y                    | Y                    |
| **4.16**                                                         | N           | N           | Y           | Y                    | Y                    |
| **4.17**                                                         | N           | N           | N           | Y                    | Y                    |
| **(Future) 4.18**                                                | N           | N           | N           | N                    | Y                    |

---

### Key

- **Y**: MCE version can create the specified Hosted Cluster version.
- **N**: MCE version cannot create the specified Hosted Cluster version.

zanetworker commented:
and another table reflecting the effect of upgrades: 

| **Hosted Cluster Version (Running)** | **2.4 to 2.5** | **2.5 to 2.6** | **2.6 to 2.7** | **(Future) 2.7 to 2.8** |
|--------------------------------------|----------------|----------------|----------------|-------------------------|
| **4.14**                             | Keep           | Keep           | Keep           | Keep                    |
| **4.15**                             | -              | Keep           | Keep           | Keep                    |
| **4.16**                             | -              | -              | Keep           | Keep                    |
| **4.17**                             | -              | -              | -              | Keep                    |
| **(Future) 4.18**                    | -              | -              | -              | -                       |

- **Keep**: The Hosted Cluster continues to run as expected during the MCE upgrade.

zanetworker commented:
We also need to reflect feature level support (e.g., disconnected, GPU, or any TP feature here)

lahinson commented:
> and another table reflecting the effect of upgrades:
> 
> **Hosted Cluster Version (Running)**	**2.4 to 2.5**	**2.5 to 2.6**	**2.6 to 2.7**	**(Future) 2.7 to 2.8**
> **4.14**	Keep	Keep	Keep	Keep
> **4.15**	-	Keep	Keep	Keep
> **4.16**	-	-	Keep	Keep
> **4.17**	-	-	-	Keep
> **(Future) 4.18**	-	-	-	-
> * **Keep**: The Hosted Cluster continues to run as expected during the MCE upgrade.

@zanetworker Can we say more here about what happens in the cases where the HyperShift Operator version does not support the hosted cluster version? Will the hosted cluster stop?

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,138 @@
+.Affects of {mce-short} updates on hosted clusters
+[cols="4",options="header"]
+|===
+|Hosted cluster version| Update from {mce-short} 2.4 to 2.5 |Update from {mce-short} 2.5 to 2.6 |Update from {mce-short} 2.6 to 2.7
+
+|4.14
+|No affect, hosted cluster continues to run
+|No affect, hosted cluster continues to run
+|No affect, hosted cluster continues to run
+
+|4.15
+|Hosted cluster is affected
+|No affect, hosted cluster continues to run
+|No affect, hosted cluster continues to run
+
+|4.16
+|Hosted cluster is affected
+|Hosted cluster is affected
+|No affect, hosted cluster continues to run
+
+|4.17
+|Hosted cluster is affected
+|Hosted cluster is affected
+|Hosted cluster is affected
+|===
```
The idea to communicate here is that as long as the HO shipped with the MCE version being upgraded to can support the version of the HostedCluster, the cluster can still be lifecycled (nothing would happen). 

So I recommend removing "Hosted cluster is affected" 

For example, upgrading from 2.4 to 2.5 while having a 4.14 hosted cluster on the management cluster will not impact the existing 4.14 cluster. Why? because the Hypershift operator shipped with MCE 2.5 can lifecycle both 4.15 AND 4.14 clusters. 

Hope that helps.

lahinson commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,138 @@
+
+[cols="2",options="header"]
+|===
+|Feature |Support level
```
I added a few more features (tolerations, GPU, HCP on ARM64 clusters on AWS).

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,144 @@
+[id="hcp-support-matrix"_{context}"]
+= Support matrix for {hcp}
+
+Because {mce} includes the HyperShift Operator, releases of {hcp} align with releases of {mce-short}. For more information, see "OpenShift Operator Life Cycles".
```
Should "OpenShift Operator Life Cycles" be a hyperlink?

lahinson replied:
We include a link to it in the Additional resources section

zanetworker replied:
Can we say we do somehow, as is I wouldn't know.

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,144 @@
+
+Because {mce} includes the HyperShift Operator, releases of {hcp} align with releases of {mce-short}. For more information, see "OpenShift Operator Life Cycles".
+
+Any supported {product-title} cluster can be a management cluster. The following table maps {mce-short} versions to the management cluster versions that support them:
```
Just a tiny nuance, let's add "standalone" before "OpenShift" for now.

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,144 @@
+|2.7
+|===
+
+For hosted clusters, no direct relationship exists the management cluster version and the hosted cluster version. The hosted cluster version depends on the HyperShift Operator of your {mce-short} version. The following table maps {mce-short} versions to which hosted cluster versions you can create by using the HyperShift Operator that is associated with that version of {mce-short}:
```
```suggestion
For hosted clusters, no direct relationship exists the management cluster version and the hosted cluster version. The hosted cluster version depends on the HyperShift Operator shipped with your {mce-short} version. The following table maps {mce-short} versions to which hosted cluster versions you can create by using the HyperShift Operator that is associated with that version of {mce-short}:
```

zanetworker replied:
or something else that reflects that HO is part of MCE maybe..

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,144 @@
+|Yes
+|===
+
+When you update to another version of {mce-short}, your hosted cluster can continue to run if the HyperShift Operator that is included in the version of {mce-short} supports the hosted cluster version. The following table shows the affects of {mce-short} updates on various versions of hosted clusters:
```
```suggestion
When you update to another version of the {mce-short}, your hosted cluster can continue to run if the HyperShift Operator that is included in the version of {mce-short} supports the hosted cluster version. The following table shows the affects of {mce-short} updates on various versions of hosted clusters:
```

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,144 @@
+|===
+|Feature |Support level
+
+|{hcp-capital} on bare metal in a disconnected environment
+|Generally Available
+
+|{hcp-capital} on {VirtProductName} in a disconnected environment
+|Generally Available
+
+|{hcp-capital} on {ibm-z-title} in a disconnected environment
+|Technology Preview
+
+|Custom taints and tolerations for {hcp} on {VirtProductName}
+|Technology Preview
+
+|NVIDIA GPU devices on {hcp} for {VirtProductName}
+|Technology Preview
+
+|{hcp-capital} for ARM64 {product-title} clusters on {aws-short}
+|Generally Available
+|===
```
Tiny nit, can we order by status (i.e., all the GAed features first then TP). An alternative would be to only highlight tech-preview features. For example, "the following features are still TP" if we want to avoid chasing existing GAed features.

lahinson replied:
Makes sense. I like the idea of listing just the TP features.

lahinson commented:
@zanetworker I incorporated your latest feedback and added a few subheadings to hopefully make the page easier to scan. Let me know what you think when you can. Thanks!

zanetworker commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,140 @@
+.Required {product-title} versions for platforms
+[cols="3",options="header"]
+|===
+|Platform |Management cluster version |Hosted cluster version
```
```suggestion
|Hosted Cluster Platform |Management cluster version |Hosted cluster version
```

zanetworker replied:
I was thinking to decouple the platform table. We would like to move to a model the management cluster can be any supported platform, and the hosted cluster platform one of the supported HCP platforms. 

The exception I can think of is for AWS: if the management cluster is on-prem, there is no way to do private-link or create private clusters.

zanetworker reviewed (approved):
lgtm :)

bscott-rh commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,140 @@
+
+
+:_mod-docs-content-type: CONCEPT
+[id="hcp-support-matrix"_{context}"]
```
```suggestion
[id="hcp-support-matrix_{context}"]
```

lahinson replied:
Good catch!

bscott-rh commented on modules/hcp-support-matrix.adoc:
```diff
@@ -0,0 +1,140 @@
+[id="hcp-matrix-updates_{context}"]
+== Updates of {mce-short}
+
+When you update to another version of the {mce-short}, your hosted cluster can continue to run if the HyperShift Operator that is included in the version of {mce-short} supports the hosted cluster version. The following table shows the affects of {mce-short} updates on various versions of hosted clusters:
```
```suggestion
When you update to another version of the {mce-short}, your hosted cluster can continue to run if the HyperShift Operator that is included in the version of {mce-short} supports the hosted cluster version. The following table shows the effects of {mce-short} updates on various versions of hosted clusters:
```
Same fix for the other uses of "affect" instead of "effect" in this file

bscott-rh replied:
This table is a bit unclear to me. I'm not sure what to make of the empty rows, should those have some other text in them like "hosted cluster can't run" or "doesn't apply" or something?

zanetworker replied:
Does not apply since for example 2.4 cant create future versions of hosted clusters so no impact of upgrade of versions that do not exist.

zanetworker replied:
Thanks Laura, looks good :)

bscott-rh reviewed (approved):
Two fixes and a question for your consideration

lahinson commented:
/cherrypick enterprise-4.17

lahinson commented:
/cherrypick enterprise-4.18
